data/database.js

```diff
@@ -38,63 +38,114 @@ class Database {
-      if (audioType === 'full' && fileLocation.full_audio_location) {
-        return fileLocation.full_audio_location;
-      } else if (
-        audioType === 'summary' &&
-        fileLocation.summary_audio_location
-      ) {
-        return fileLocation.summary_audio_location;
-      }
-    }
-    return '';
+  async getAudioFileLocation(articleId, type, voice) {
+    logger.info(`getAudioFileLocation for ${articleId}/${type}`);
+    return new Promise(resolve => {
+      AudioFiles.query('item_id')
+        .eq(articleId)
+        .filter('type')
+        .eq(type)
+        .filter('voice')
+        .eq(voice)
+        .filter('codec')
+        .eq('mp3')
+        .exec()
+        .then(function(data) {
+          console.log(data);
+          console.log(JSON.stringify(data));
+          if (data.count) {
+            resolve(data[0].url);
+            if (data.count > 1) {
+              logger.warn('duplicate entries!!!');
+            }
+          } else {
+            resolve('');
+          }
+        });
+    });
   }
 
-  async getMobileFileLocation(articleId) {
-    logger.info(`getMobileFileLocation for ${articleId}`);
-    const fileLocation = await AudioFileLocation.get({ item_id: articleId });
-    if (fileLocation && fileLocation.mobile_audio_location) {
-      return fileLocation.mobile_audio_location;
-    }
-    return '';
+  async getMobileMetadata(articleId) {
+    logger.info(`getMobileMetadata for ${articleId}`);
+    return new Promise(resolve => {
+      AudioFiles.query('item_id')
+        .eq(articleId)
+        .filter('type')
+        .eq('mobile')
+        .exec()
+        .then(function(data) {
+          console.log(data);
+          console.log(JSON.stringify(data));
+          resolve(data);
+        });
+    });
   }
 
-  async storeAudioFileLocation(articleId, audioType, location) {
-    logger.info(
-      `storeAudioFileLocation for ${articleId}/${audioType}: ${location}`
-    );
-    let fileLocation = await AudioFileLocation.get({ item_id: articleId });
-    if (!fileLocation) {
-      fileLocation = new AudioFileLocation({
-        item_id: articleId
-      });
-    }
-    if (audioType === 'full') {
-      fileLocation.full_audio_location = location;
-      fileLocation.full_audio_date = Date.now();
-    } else if (audioType === 'summary') {
-      fileLocation.summary_audio_location = location;
-      fileLocation.summary_audio_date = Date.now();
-    }
-    await fileLocation.save();
+  async storeAudioFileLocation(articleId, type, voice, location) {
+    logger.info(`storeAudioFileLocation for ${articleId}/${type}: ${location}`);
+    let mp3 = new AudioFiles({
+      item_id: articleId,
+      uuid: uuidgen.generate(),
+      voice: voice,
+      codec: 'mp3',
+      bitrate: 40000,
+      samplerate: 16000,
+      type: type,
+      url: location,
+      date: Date.now()
+    });
+    await mp3.save();
+
+    logger.debug('Before opus save');
+    let opus = new AudioFiles({
```

Comment:
I'm a little uncomfortable creating the opus records here when we know that the file doesn't exist yet. one addition that would be nice is to have the "processing" status stored here in the AudioFiles table when we create the record, and then let scout-xcode inform scout-ua when it has been completed. that way we know from the data directly which of these files are real and which are still in process (or never got processed)

Reply:
I think we can put the status and let scout-xcode update the database... we'd have to add more to the payload of the SQS message in order for scout-xcode to figure it out, but I think I'd rather not create antoher channel back to scout-ua.  I'll file a follow up for this one.

Reply:
https://github.com/MozScout/scout-ua/issues/158